test/lint.js

```diff
@@ -141,13 +138,10 @@ const checkFiles = (...files) => {
  * @returns {boolean} Whether there were any errors
  */
 const main = (files) => {
-  /** @type {boolean} */
-  var hasErrors = false;
-
   if (files) {
```

Comment:
If this function is called twice, the second call has the error from the previous call. So it would be good to reset `errors` and `fileErrors` here as well, even if `main` is normally only called once.

```
errors = [];
fileErrors = 0;
```

(The alternative would be pass a result object into `checkFiles()` to avoid the global variables altogether.)